02-clients/01-studio.md

```diff
@@ -0,0 +1,424 @@
+   transactions.
+5. [**explain**] button — enable [explanations](../0001-typedb/02-dev/06-infer.md#explain-query) mechanics by providing 
+   `explainables` methods for results. Only available for read transactions with **snapshot** and **infer** enabled.
+6. [**Transaction status**] (Circle (●) icon) indicator — transaction activity indicator. Green if there is a 
+   transaction opened by TypeDB Studio at this moment. Gray if there is no opened transaction. Rotating animation — 
+   TypeDB Studio processing the query or the results (including additional API calls).
+7. [**Close transaction**] (x icon) button — close active transaction without committing the results.
+8. [**Rollback transaction**] (U-turn (⟲) icon) button — revert the transaction results without closing it.
+9. [**Commit transaction**] (Tick (✔) icon) button — commit the changes and close the transaction.
+10. [**Run Query**] (play (▶) icon) button — run query or queries in the active tab of the [**Text editor**] panel.  
+    It will open a transaction (with the session type and transaction type specified in the [**Toolbar**] on the left)  
+    and execute the TypeQL code against the selected database.
+11. [**Stop Signal**] (lighting (⚡) icon) button — stop query execution after the next result.
+
+##### Run query
+
+Make sure TypeDB Studio [connected](#connect-to-typedb-server) to a TypeDB Server, 
+database [selected](#select-active-database) and project directory [opened](#open-a-project-directory).
+
+1. [Write a new](#create-a-new-file) or [open an existing](#open-an-existing-file) file with query to run.
+
+   ![Query text](../images/studio/project-schema-pasted.png)
+
+2. Choose the correct session (`schema`/`data`) type and transaction (`read`/`write`) type to run the query by  
+   clicking on those options on the [**Toolbar**] switches.
+
+3. Run query by clicking on the [**Run Query**] (▶) button. [**Run**] panel will be expanded on the bottom of the  
+   query to show the results of the query execution attempt.
+
+   ![Write the Schema](../images/studio/project-schema-query-run.png)
+
+4. If it's the `write` transaction type — wait for transaction control options to be ready and commit or close the 
+   transaction (by clicking on the [**Commit transaction**] (✔) or [**Close transaction**] (x) buttons respectively).
+
+The result of the query is displayed in the [**Run**] panel under the Text editor.
+
+If any error occurs during the query validation or transaction it is shown as red popup in the bottom right corner of 
+the TypeDB Studio window. Additional information about the error is displayed in the results of the query in the 
+[**Run**] panel.
+
+If a `write` transaction committed successfully — a blue pop-up with a transaction confirmation is displayed. See 
+example below.
+
+![Commit the Transaction](../images/studio/project-schema-committed.png)
+
+## Types explorer capabilities
```

Comment:
We call this the 'Type Browser' in our code, so let's reflect this here too.

Reply:
I see your argument. I think we should create an issue specific to the nomenclature of Studio components and come to agreement on all of thenaming that needs to be changed.

In the mean time, I think it's important that we don't enter a state where the source code and the documentation diverge in how things are named. It's a tenant of our design philosophy that we have a ubiquitous language (at least per product) that we adhere to across documentation, code and discussions.

Reply:
1. We have discussed the approach with Shane and I think he is more flexible with this. We are re-implementing our external communications framework and rewriting the documentation. It is time to change the terminology now if we think it will help with our external communications. The only question is what will be chosen as the right terminology.
2. There is a meeting about terminology planned. I'm waiting for Shane to approve it before proceeding.